src/common/utils/iter.ts

```diff
@@ -236,3 +238,13 @@ export function every<T>(src: Iterable<T>, fn: (val: T) => any): boolean {
 
   return true;
 }
+
+export function* concat<T>(src1: IterableIterator<T>, src2: IterableIterator<T>): IterableIterator<T> {
```

Comment:
Isn't it starting to get at least a bit scary to produce integral technical functions without accompanying technical unit tests? :)

Reply:
Sure I can do that.